bitnami/matomo/templates/NOTES.txt

```diff
@@ -0,0 +1,86 @@
+
+{{- else -}}
+
+########################################################################################
+### ERROR: You did not provide an external database host in your 'helm install' call ###
+########################################################################################
+
+This deployment will be incomplete until you configure Matomo with a resolvable database
+host. To configure Matomo to use and external database host:
+
+1. Complete your Matomo deployment by running:
+
+  export APP_PASSWORD=$(kubectl get secret --namespace {{ include "common.names.namespace" . }} {{ template "matomo.secretName" . }} -o jsonpath="{.data.matomo-password}" | base64 -d)
+
+  ## PLEASE UPDATE THE EXTERNAL DATABASE CONNECTION PARAMETERS IN THE FOLLOWING COMMAND AS NEEDED ##
+
+  helm upgrade --namespace {{ include "common.names.namespace" . }} {{ .Release.Name }} bitnami/{{ .Chart.Name }} \
+    --set matomoPassword=$APP_PASSWORD,service.type={{ .Values.service.type }},mariadb.enabled=false{{- if not (empty .Values.externalDatabase.user) }},externalDatabase.user={{ .Values.externalDatabase.user }}{{- end }}{{- if not (empty .Values.externalDatabase.password) }},externalDatabase.password={{ .Values.externalDatabase.password }}{{- end }}{{- if not (empty .Values.externalDatabase.database) }},externalDatabase.database={{ .Values.externalDatabase.database }}{{- end }},externalDatabase.host=YOUR_EXTERNAL_DATABASE_HOST{{- if .Values.global }}{{- if .Values.global.imagePullSecrets }},global.imagePullSecrets={{ .Values.global.imagePullSecrets }}{{- end }}{{- end }}
+
+{{- end }}
+
+{{- include "common.warnings.rollingTag" .Values.image }}
+{{- include "common.warnings.rollingTag" .Values.metrics.image }}
+{{- include "common.warnings.rollingTag" .Values.volumePermissions.image }}
+{{- include "common.warnings.rollingTag" .Values.certificates.image }}
+
+{{- $passwordValidationErrors := list -}}
+{{- if not .Values.existingSecret -}}
+    {{- $secretName := include "matomo.secretName" . -}}
+    {{- $requiredMatomoPassword := dict "valueKey" "matomoPassword" "secret" $secretName "field" "matomo-password" "context" $ -}}
+    {{- $requiredMatomoPasswordError := include "common.validations.values.single.empty" $requiredMatomoPassword -}}
+    {{- $passwordValidationErrors = append $passwordValidationErrors $requiredMatomoPasswordError -}}
+{{- end -}}
+
+{{- $mariadbSecretName := include "matomo.databaseSecretName" . -}}
+{{- $mariadbPasswordValidationErrors := include "common.validations.values.mariadb.passwords" (dict "secret" $mariadbSecretName "subchart" true "context" $) -}}
+{{- $passwordValidationErrors = append $passwordValidationErrors $mariadbPasswordValidationErrors -}}
+
+{{- include "common.errors.upgrade.passwords.empty" (dict "validationErrors" $passwordValidationErrors "context" $) -}}
```

Comment:
MariaDB chart is using `common.secrets.passwords.manage` helper to avoid this kind of validations. 

You can use that helper in your secret to retrieve the existing password from the secret during upgrades